src/duckling/game/path/path-drawer.ts

```diff
@@ -0,0 +1,168 @@
+    }
+    
+    if (!pathAttribute.isLoop) {
+        return _drawPath(pathAttribute.vertices);
+    } else {
+        return _drawLoopPath(pathAttribute.vertices);
+    }
+}
+
+function _drawLoopPath(vertices : Vector[]) : DisplayObject {
+    let container = new Container();
+    for (let i = 1; i < vertices.length - 1; i++) {
+        let startVertex = vertices[i];
+        let endVertex = vertices[i + 1];
+        container.addChild(_drawLine(startVertex, endVertex));
+        container.addChild(_drawJoint(endVertex.x, endVertex.y));
+    }
+    container.addChild(_drawLine(vertices[vertices.length - 1], vertices[0]));
+    
+    // draw the first loop last so the indicator will be drawn on top of other segments
+    container.addChild(_drawFirstLoopSegment(vertices[0], vertices[1]));
+    return container;
+}
+
+function _drawFirstLoopSegment(startVertex : Vector, endVertex : Vector) : DisplayObject {
+    let container = new Container();
+    let rotatedEndVertex = vectorRotate(endVertex, -_pathTheta(startVertex, endVertex), startVertex);
+    
+    container.addChild(_drawLine(startVertex, rotatedEndVertex));
+    container.addChild(_drawJoint(rotatedEndVertex.x, rotatedEndVertex.y));
+    container.addChild(_drawSquare(startVertex.x, startVertex.y, STARTING_SQUARE_COLOR));
+    
+    let loopDirectionPosition = { x: startVertex.x + DIRECTION_INDICATOR_OFFSET, 
+                                  y: startVertex.y };
+    container.addChild(_drawRightTriangle(loopDirectionPosition, STARTING_SQUARE_COLOR));
+
+    return _rotateSegemnt(container, _pathTheta(startVertex, endVertex), startVertex);
+}
+
+function _drawPath(vertices : Vector[]) : DisplayObject {
+    let container = new Container();
+    container.addChild(_drawSegmentWithLeftSquare(vertices[0], vertices[1]));
+    container.addChild(_drawJoint(vertices[1].x, vertices[1].y));
+    for (let i = 1; i < vertices.length - 2; i++) {
+        let startVertex = vertices[i];
+        let endVertex = vertices[i + 1];
+        container.addChild(_drawLine(startVertex, endVertex));
+        container.addChild(_drawJoint(endVertex.x, endVertex.y));
+    }
+    container.addChild(_drawSegmentWithRightSquare(vertices[vertices.length - 2], vertices[vertices.length - 1]));
+    return container;
+}
+
+function _drawSegmentWithBothSquares(startVertex : Vector, endVertex : Vector) : DisplayObject {
+    return _drawSegmentWithSquares(startVertex, endVertex, true, true);
+}
+
+function _drawSegmentWithLeftSquare(startVertex : Vector, endVertex : Vector) : DisplayObject {
+    return _drawSegmentWithSquares(startVertex, endVertex, true, false);
+}
+
+function _drawSegmentWithRightSquare(startVertex : Vector, endVertex : Vector) : DisplayObject {
+    return _drawSegmentWithSquares(startVertex, endVertex, false, true);
+}
+
+function _drawSegmentWithSquares(startVertex : Vector, endVertex : Vector, leftSquare : boolean, rightSquare : boolean) : DisplayObject {
```

Comment:
Nah, what I mean is that You wouldn't have any flavor of drawSegmentWithSquare. Instead, where you call those different functions, you would just call drawSegment, then call drawSquare